src/imu/imu.cpp

```diff
@@ -65,6 +73,14 @@ bool getAccelerometer(float *acc_x, float *acc_y, float *acc_z)
 bool getGyroscope(float *gyr_x, float *gyr_y, float *gyr_z)
 {
 #if SF_PLATFORM == SF_PLATFORM_PARTICLE
+
+    myICM.getAGMT();
+
+    if (myICM.status != ICM_20948_Stat_Ok)
+    {
+        FLOG_AddError(FLOG_ICM_FAIL, 0);
+        return false;
+    }
     ICM_20948_AGMT_t agmt = myICM.getAGMT();
```

Comment:
Lines 77 and 84 appear to have duplicate behavior.  Would it be simpler to move line 84 earlier?